signing_service/src/signing_service.py

```diff
@@ -126,6 +127,40 @@ def _validate_arguments(self):
 
 
 def _parse_arguments() -> argparse.Namespace:
+
+    def get_secret_variable_wrapper(env_variable_name=None, file=False, base64_convert=False):
+        def wrapper(**kwargs):
+            return GetSecretVariable(env_variable_name, file, base64_convert, **kwargs)
+
+        return wrapper
+
+    class GetSecretVariable(argparse.Action):
+        def __init__(self, env_variable_name, file, base64_convert, option_strings, dest, nargs=None):
+            self.env_variable_name = env_variable_name
+            self.file = file
+            self.base64_convert = base64_convert
+            if self.env_variable_name:
+                nargs = 0
+            super(GetSecretVariable, self).__init__(option_strings=option_strings, dest=dest, nargs=nargs)
+
+        def __call__(self, parser, namespace, values, option_string=None):
+            if values and self.file:
+                try:
+                    with open(values) as file:
+                        self.const = file.read()
+                except FileNotFoundError as file_not_found_error:
+                    crash_logger.error(f'{file_not_found_error}')
+                    raise
+            elif values:
+                self.const = values
+            elif self.env_variable_name:
+                self.const = os.environ.get(self.env_variable_name)
```

Comment:
There's no `else`. What happens if none of these conditions is true?

If it's not possible then you should have `assert False` in that `else`.